### ISSUE TITLE
Prevent usage of `return`, `yield` in `try/finally`

### ISSUE THREAD

**Issue by hofrob** [pylint-dev/pylint#8260]
### Current problem

Code in `finally` blocks will always be executed. If there's a `return` inside the `try`, it's possible to overwrite this return value with a `return` in the `finally`.

```py
def add(a: int, b: int) -> int:
    try:
        return a + b
    finally:
        return 0


if __name__ == "__main__":
    print(add(1, 2))  # those args will be ignored; "0" is printed
```

### Desired solution

There should not be any `return` or `yield` statements in a finally block.

### Additional context

https://docs.python.org/3/reference/compound_stmts.html#finally-clause

> The return value of a function is determined by the last return statement executed.

**Comment by nickdrozd** [comment]
Sounds like a good idea to me. I did a quick search of a variety of repos and I couldn't find a single instance, so it seems to be a rare pattern.

**Comment by deepgohil** [comment]
I would like to solve this issue.

**Comment by Pierre-Sassoulas** [comment]
I assigned you to it @deepgohil :)

**Comment by deepgohil** [comment]
Can you tell me where I have to change the code? I want to know folder name

**Comment by Pierre-Sassoulas** [comment]
You can create or modify a checker in ``pylint/checkers``, or create and modify an extension in ``pylint/extensions``. This check feel like a nasty underhanded bug that could be a default checker of warning level imo.

Design by ChatGPT would indicate that it could be in the basic checker:

>> In what pylint checker should a check for 'return-superseded-by-finally' should go ?

> The return-superseded-by-finally check in Pylint belongs to the "basic" category of checks. The "basic" checks are a set of checks that Pylint performs by default, without the need for any additional configuration. This check reports a warning when a return statement appears in a function that also has a finally block.

But I think that ChatGPT is basic and might be giving bad design advise here. I don't know where it should really go though. Or if it should really be called 'return-superseded-by-finally'.

**Comment by ollie-iterators** [comment]
What should be in the finally block instead of return?

**Comment by Pierre-Sassoulas** [comment]
> What should be in the finally block instead of return?

The finally block should be removed for example like this:
```python
def add(a: int, b: int) -> int:
    try:
        return a + b
    except:
        return 0
```        
 @deepgohil do you have an opinion regarding the name and the checker it should go in ? (maybe a new one ?)

**Comment by hofrob** [comment]
I'd say in general: `finally` blocks should not contain any `yield` or `return` statements. I would not make any suggestion on how to fix it. 

I'm also wondering where to put this new rules :slightly_smiling_face:

**Comment by hofrob** [comment]
Since nobody is working on this, I'd try to come up with something. Let's see how this goes.
